### ISSUE TITLE
Distinguishing between missing credentials and incorrect credentials in app.py 

### ISSUE THREAD

**Issue by py-ranoid** [HTTP-APIs/hydrus#152]
### I'm submitting a 
-  feature request.

### Current Behaviour:
failed_authentication() is called when credentials are missing as well as when credentials are incorrect. 

### Expected Behaviour:
Add a parameter `missing : bool` to `failed_authentication()` to return (when credentials are missing)
```
{
  "401": "Need credentials to authenticate."
}
```
and (when credentials are incorrect)
```
{
   "403": "Credentials were incorrect."
}
```
### Do you want to work on this issue?
Yes

**Comment by Mec-iS** [comment]
Is this been implemented with the merged Authentication system at #144 ?

**Comment by Ankur-singh** [comment]
hey @py-ranoid !
can I work on this, if its still open ..??
This is my first time, any suggestion is welcome.

**Comment by chrizandr** [comment]
It has been implemented in #144
